docs/PERMISSION_SETTING.md

```diff
@@ -0,0 +1,68 @@
+# Permission setting for non-root users' usage of Remoteproc Runtime
+
+For non-root users to use Remoteproc Runtime, the remoteproc driver and the container engine must be accessible for this user.
+
+## How to set Remoteproc Runtime to be accessible by non-root users
+
+### 1. Make remoteproc driver accessible to the user
+
+Usually, remoteproc driver can only be accessible to root. To change this setting, follow the below instructions:
+
+1. Create a group and add your user:
+
+   ```
+   sudo groupadd remoteproc
+   sudo usermod -aG remoteproc "$USER"
+   ```
+
+   Log out and log back in to refresh group membership
+
+2. Use systemd-tmpfiles to set mode/owner on every boot (and re-apply easily):
+
+   Create /etc/tmpfiles.d/remoteproc.conf with following:
+
+   ```
+   f /sys/class/remoteproc/remoteproc0/state                0664  root remoteproc -   -
+   f /sys/class/remoteproc/remoteproc0/firmware             0664  root remoteproc -   -
+   f /sys/class/remoteproc/remoteproc0/name                 0664  root remoteproc -   -
+   ```
+
+   Add similar lines for each additional remoteproc device (e.g., remoteproc1, remoteproc2, etc.) as needed.
+
+3. Apply the change in remoteproc.conf using root permission:
+   ```
+   sudo systemd-tmpfiles --create /etc/tmpfiles.d/remoteproc.conf
+   ```
+4. Log in as a user in the remoteproc group and try the following commands to make sure that you can access the remoteproc driver as this user:
+   ```
+   # read state
+   cat /sys/class/remoteproc/remoteproc0/state
+   # start/stop
+   echo start | tee /sys/class/remoteproc/remoteproc0/state
+   echo stop  | tee /sys/class/remoteproc/remoteproc0/state
+   ```
+
+### 2. Make sure the user's systemd instance is alive
```

Comment:
Why is `systemd` necessary as part of this?

Reply:
`/sys/class` permissions are not persistent across reboot as these are virtual filesystem entries. However your point stands that we should seek to have a light touch with documenting technologies which aren't a core part of our offering.